best-practices.md

```diff
@@ -68,21 +60,40 @@ See more
 
-Most javascript libraries rely on `addEventListener`, `setTimeouts` and others that trigger change detection in angular2. 
-Triggering, or listening to events of 3rd party libs can trigger excesively change detection decreasing considerably the perfomance of the application.
+Please also keep in mind, that if you use Observable.create and then hoist the input out of the create call, you have just made a subject.
 
-On the other hand, these libraries can rely on event handling not patched by angular2, therefore these events will run outside angular2's zone causing change detection NOT to run!
+### What should I use instead?
 
-If some cool library hasn't been ported to angular2, be the first to port it! Make yourself famous!
+First and foremost, if you already have an Observable, or something that can easily be observed (Controls, Events, etc..) you should just observe it, and extend the processing chain to meet your needs (making it cold or hot, etc..).
+
+Once you have exhausted all of your options for Obervable.x besides create and Subject, then you must if your entire sequence can be determined at creation. If this is the case, then use Observable.create, otherwise use a Subject.
 
+Further Reading:
+- (http://tomstechnicalblog.blogspot.ca/2016/03/rxjava-problem-with-subjects.html)
+- (http://davesexton.com/blog/post/To-Use-Subject-Or-Not-To-Use-Subject.aspx)
+- (https://medium.com/@benlesh/learning-observable-by-building-observable-d5da57405d87#.12a0rfxtt)
+- (https://medium.com/@benlesh/hot-vs-cold-observables-f8094ed53339#.x0q63ntrm)
 
-TODO : Improve more ?
 
+## Don't mix Angular 2 with third party libraries
+
+Angular 2 works far better with third party libraries than Angular 1 did. In many cases, a library will work just fine with Angular 2. The biggest reasons not to use third party libraries is that they simply will not work optimally with Angular 2.
+
+Most javascript libraries rely on `addEventListener`, `setTimeouts` and others that trigger change detection in Angular 2. Triggering, or listening to events of 3rd party libs can trigger excesive change detection, and considerably decrease the perfomance of your application.
+
+On the other hand, these libraries can rely on event handling not patched by angular2, therefore these events will run outside angular2's zone causing change detection NOT to run!
+
+Additionally, Angular 2 wants templates to be compiled at build time, and not include the compiler in your redistributable (improving speed and load time). This means that while it is possible to use third party libraries in your templates, it is not possible to compile angular templates in your third party libs, as that would require exposing the compiler and bundling it.
+
+With all of that said, libraries that don't manipulate the dom, and don't need to be patched to work with zones can and do work perfectly with Angular 2, without patching or modification.
+

```

Comment:
I've had success with 3rd party libs that do manipulate the DOM so this may be misleading but worth describing the approach: use `@Input()` and OnPush change detection strategy, handle wireup in ngAfterViewInit and tear down in destroy etc... (I'll try and create a proper example)